android/app/src/main/java/com/on/turip/data/common/TuripResultExtensions.kt

```diff
@@ -0,0 +1,75 @@
+package com.on.turip.data.common
+
+/**
+ * 성공일 경우
+ */
+inline fun <T> TuripCustomResult<T>.onSuccess(action: (value: T) -> Unit): TuripCustomResult<T> {
+    if (this is TuripCustomResult.Success) action(value)
+    return this
+}
+
+/**
+ * 예외 발생한 경우
+ */
+inline fun <T> TuripCustomResult<T>.onFailure(action: (errorType: ErrorType) -> Unit): TuripCustomResult<T> {
+    if (this is TuripCustomResult.Failure) action(errorType)
+    return this
+}
+
+/**
+ * 예외 발생한 경우(로깅용)
+ */
+inline fun <T> TuripCustomResult<T>.onFailureWithCause(action: (errorType: ErrorType, cause: Throwable?) -> Unit): TuripCustomResult<T> {
+    if (this is TuripCustomResult.Failure) action(errorType, cause)
+    return this
+}
+
+/**
+ * 안전한 변환 지원
+ */
+inline fun <R, T> TuripCustomResult<T>.mapCatching(transform: (value: T) -> R): TuripCustomResult<R> =
+    when (this) {
+        is TuripCustomResult.Success -> {
+            runCatching { transform(value) }.fold(
+                onSuccess = { TuripCustomResult.Success(it) },
+                onFailure = { TuripCustomResult.Failure(ErrorType.Unknown, cause = it) },
+            )
+        }
+
+        is TuripCustomResult.Failure -> {
+            this
+        }
```

Comment:
이런 건 
```suggestion
        is TuripCustomResult.Failure -> this
```

이렇게 써도 되지 않나염 ?

Reply:
반영 커밋 : https://github.com/woowacourse-teams/2025-Turip/pull/508/commits/ae5009ea92596baf7934482d185a3b598efa480e